lib/plugin/instance.rb

```diff
@@ -239,6 +239,17 @@ def register_editable_group_custom_field(field)
     DiscoursePluginRegistry.register_editable_group_custom_field(field, self)
   end
 
+  # Allows to define custom filter utilizing the user's input.
+  # Ensure proper input sanitization before using it in a query.
+  #
+  # Example usage:
+  #   register_custom_filter("word_count") do |scope, value|
+  #     scope.where(word_count: value)
+  #   end
+  def register_custom_filter(filter_name, &block)
+    TopicsFilter.add_filter(filter_name, &block)
+  end
```

Comment:
I think we should make this auto-disable whenever the plugin is disabled. Probably best to lean on DiscoursePluginRegistry for that.